x-pack/plugins/xpack_main/server/lib/populate_ui_capabilities.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { UICapabilities } from 'ui/capabilities';
+import { Feature } from '../../types';
+
+export function populateUICapabilities(
+  xpackMainPlugin: Record<string, any>,
+  injectedVars: Record<string, any>
+): UICapabilities {
+  const features: Feature[] = xpackMainPlugin.getFeatures();
+
+  const featureCapabilities: UICapabilities[] = features.map(getCapabilitiesFromFeature);
+
+  return mergeCapabilities(injectedVars.uiCapabilities || {}, ...featureCapabilities);
+}
+
+function getCapabilitiesFromFeature(feature: Feature): UICapabilities {
+  const capabilities: UICapabilities = {
+    navLinks: {},
+    [feature.id]: {},
+  };
+
+  if (feature.navLinkId) {
```

Comment:
I could go either way here -- I know that OSS currently populates all known nav links, so this is probably duplicative/unnecessary. I put it here for completeness, more as a defensive measure than anything else.

Reply:
Hmmm... I think I'd prefer we remove the duplication and let OSS handle this.